control-plane-operator/controllers/hostedcontrolplane/mcs/reconcile.go

```diff
@@ -0,0 +1,143 @@
+package mcs
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/hypershift/control-plane-operator/api"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/util"
+	mcfgv1 "github.com/openshift/hypershift/thirdparty/machineconfigoperator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+func ReconcileMachineConfigServerConfig(cm *corev1.ConfigMap, p *MCSParams) error {
+	p.OwnerRef.ApplyTo(cm)
+	if cm.Data == nil {
+		cm.Data = map[string]string{}
+	}
+	serializedDNS, err := serialize(p.DNS)
+	if err != nil {
+		return err
+	}
+	serializedInfra, err := serialize(p.Infrastructure)
+	if err != nil {
+		return err
+	}
+	serializedNetwork, err := serialize(p.Network)
+	if err != nil {
+		return err
+	}
+	serializedProxy, err := serialize(p.Proxy)
+	if err != nil {
+		return err
+	}
+	serializedPullSecret, err := serialize(p.PullSecret)
+	if err != nil {
+		return err
+	}
+	serializedMasterConfigPool, err := serializeConfigPool(masterConfigPool())
+	if err != nil {
+		return err
+	}
+	serializedWorkerConfigPool, err := serializeConfigPool(workerConfigPool())
+	if err != nil {
+		return err
+	}
+
+	cm.Data["root-ca.crt"] = string(p.RootCA.Data[pki.CASignerCertMapKey])
```

Comment:
Do we allow adding more stuff into this? As written right now, we keep additional keys

Reply:
we should not, will fix